src/style/style.js

```diff
@@ -85,65 +127,78 @@ class Style extends Evented {
             }
         });
 
-        const stylesheetLoaded = (err, stylesheet) => {
+        const stylesheetLoaded = (err, stylesheet: ?StyleSpecification) => {
```

Comment:
🤔 should we typed this as `mixed` and then cast it to `StyleSpecification` after it passes validation?